compiler/rustc_parse/src/parser/attr.rs

```diff
@@ -429,7 +430,7 @@ pub fn maybe_needs_tokens(attrs: &[ast::Attribute]) -> bool {
         if attr.is_doc_comment() {
             return false;
         }
-        attr.ident().map_or(true, |ident| {
+        attr.ident().is_none_or(|ident| {
             ident.name == sym::cfg_attr || !rustc_feature::is_builtin_attr_name(ident.name)
         })
```

Comment:
`maybe_needs_tokens` only ever seems to be used as `!maybe_needs_tokens`, so maybe this function should be inverted:

```rust
// name to be bikeshed
pub fn is_complete(attrs: &[ast::Attribute]) -> bool {
    attrs.iter().all(|attr| {
        attr.is_doc_comment() || attr.ident().is_some_and(|ident| {
            ident.name != sym::cfg_attr && rustc_feature::is_builtin_attr_name(ident.name)
        })
    })
}
```

"The attributes are complete if all tokens are either a doc comment or a builtin attribute other than cfg_attr."